osu.Game/Overlays/NowPlayingOverlay.cs

```diff
@@ -234,7 +234,7 @@ protected override void Update()
                 pendingBeatmapSwitch = null;
             }
 
-            var track = beatmap.Value?.TrackLoaded ?? false ? beatmap.Value.Track : null;
+            var track = musicController.TrackLoaded ? musicController.CurrentTrack : null;
```

Comment:
I'm aware this is the minimal and henceforth "safe" change, but unless I'm missing something it seems to me that you could drop this entirely and just inspect `CurrentTrack`.